rita_exit/src/database/database_tools.rs

```diff
@@ -15,6 +17,9 @@ use exit_db::{models, schema};
 use std::net::IpAddr;
 use std::net::Ipv4Addr;
 
+/// Max number of time we loop to find an available ip before we quit
+const MAX_DEPTH: u8 = 25;
```

Comment:
we should avoid recursion in production, it's unpredictable, the algorithm should assign Subnets to clients in a predictible way and allow them to be returned to the pool for later allocation.